### ISSUE TITLE
GameGuru MAX - UI Request - Indoor Level "Biom"

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2113]
Currently, GameGuru MAX seems to be heavily geared toward outdoor levels. However, there are many of us that want to also use it to create interior-only FPS games and believe this should be an option within GameGuru MAX. As a result, I would like to suggest that the editor have a few options. One would be a "biom" that is only for indoor-only purposes. Instead of "empty", it could read "indoor" or "interior level" or something akin to that. Once selected, you would not need to "generate terrain" and once in the editor, you would not get a default sky (no light, no sun, no clouds, no skycube, etc.). You would simply be in the editor, blank, and ready to develop your interior only level. Something like this would provide a quick and easy way for someone to just jump in and start creating their interior-only levels in MAX.

**Comment by LeeBamberTGC** [comment]
@ArgentArts A good idea.  I am sure some performance can be regained by switching a bunch of outdoor-centric elements off. Alas we don't have a great solution yet for building out interiors, but it is something we are thinking about for 2022. The Structure Editor did not quite cut it, but we also don't want to turn it into a full 3D modeler. Something in the middle would be good, perhaps inspired by Sims or builder-style games.  Links to videos showing good techniques would be great here, and we can continue discussing this.

**Comment by ArgentArts** [comment]
I think I can speak for many of us that want to do interior levels - we don't need anything but an empty level - no terrain, no sky. Oh, snapping on the UP axis would be nice, though. I don't need EBE or anything like that. I think many of us just intend to import our own created "blocks" (floor, wall, and ceiling models) and snap them in place or create complete levels in our 3D modeling program or purchase modular kits to use. So, for me (and, I think, many others) providing a blank slate would be a great start for interior level designers. Nothing more is really needed. 

EBE and things like that can be an addition that others may need, but not all.

**Comment by LeeBamberTGC** [comment]
@rvanner Can you add this as a post-EA item, I don't think it is a huge amount of work but we need to figure out how to incorporate it into the design, perhaps a new button in  the storyboard such as Add Empty Level, which would skip the terrain generator and create a TOTALLY EMPTY void for advanced users:

![image](https://user-images.githubusercontent.com/35840583/154117723-7aa5d89f-e1da-4c3c-bdc9-aec434c9d54c.png)

**Comment by LeeBamberTGC** [comment]
@ArgentArts Thanks for your ideas. @rvanner @ZakJudges @dllllllllllb @PaulSJ @plemsoft including the rest of the internal team to get their views too.

**Comment by ArgentArts** [comment]
You're welcome.

One of the reasons I suggested the "Click to Start" instead of the "Generate Terrain" button name is it's currently confusing the first time you ever use MAX. When you first use MAX, you get prompts - A big button that says, "Create a New Game Project", after that a big area that says, "CLICK HERE TO CREATE A NEW LEVEL". So, far, you've been guided. But once you get to the Terrain Generator, the new user is left hanging. There isn't a button that says anything remotely akin to "CLICK ME TO MOVE ON!" The first time I encountered this screen, it frustrated the heck out of me. It did others, too (from what I read on both Discord and the forum). It's easy, AFTER you know what to do. But it makes you want to throw your computer out the window BEFORE you know what to do. ;) Therefore, if the "Generate Terrain" button, being the last thing that needs to be clicked before moving on to the editor were renamed to something like "Click to Start" or "Click to Go to Editor" or "Click to Proceed" and perhaps the button where made to stand out from the rest (big, bright color), then the end-user would know that's what they need to do to move on. The mouse-over text could say something like, "Click here after setting all desired parameters to move on to the level editor."

And if there was a "biom" for an interior level here (with no need to generate a terrain at all), then this button would no to not generate a terrain if that "biom" were selected. It would do what was needed, based on the parameters selected, and would advance the end-user to the editor.

**Comment by ZakJudges** [comment]
I agree having a completely empty level would be a good addition and having it in the terrain generator would make sense. It could even be a setting that displays when the "Empty" biome is selected, to prevent having to redesign anything. Maybe a checkbox somewhere here:
![Screenshot 2022-02-15 191302](https://user-images.githubusercontent.com/32545391/154132454-f00165d5-6054-4103-a192-a78db7acbd60.png)

It would also be nice to have a way to bypass the terrain generator, going straight into a completely empty editor. But I don't think its a necessity. It might save a precious few seconds, but considering that you'll spend 99% of your time elsewhere in the software, I don't think it's too important. More of a nice to have :)

**Comment by ArgentArts** [comment]
`It would also be nice to have a way to bypass the terrain generator, going straight into a completely empty editor. But I don't think its a necessity. It might save a precious few seconds, but considering that you'll spend 99% of your time elsewhere in the software, I don't think it's too important. More of a nice to have :)`

Actually, it is important to not generate terrain if the end-user is creating an indoor-only level because there will be some performance gain by not having any terrain at all. Why have anything in the level eating up CPU/GPU/memory that's never going to be seen or used?

**Comment by ZakJudges** [comment]
@ArgentArts Yes I agree. I'm saying the place for the option to create an indoor only level should be inside the terrain generator, to keep the workflow the same, as you say. My suggestion was to put a checkbox somewhere that would switch off the terrain, sky, all of it. So you wouldn't really be eating up resources, as it would be deleted when you begin editing the level :)

**Comment by ArgentArts** [comment]
Ah! I get you! Thanks for clearing that up. I agree. My suggestion was to create an Indoor Only button, similar to the Empty Button, that would do all the work for you. One click and done. ;)

**Comment by OldFlak** [comment]
The monkey has it right :)
There is no point going to the terrain editor to create an empty level - makes no sense at all. 
Clicking a button to create an empty level that won't make use of anything in the terrain editor does not break the workflow, it **simplifies** it, and just makes sense :)

Just Have Two buttons:
1) Empty level - which simply has nothing there at all (no need for terrain editor - so why go there?)
2) Terrain level - would take you to the terrain editor to create your outside level
